### PR TITLE
Add beforeunload to dataset upload view

### PR DESCRIPTION
**Pull request by grittaweisheit**
This PR double checks with the user if he/she really wants to leave the dataset upload view. 

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- upload a dataset
- try to navigate elsewhere while it's still uploading
- you should be warned about leaving while it's still loading

### Issues:
- fixes #5051

------
- [x] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] ~~Updated [(unreleased) migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable~~
- [ ] ~~Updated [documentation](../blob/master/docs) if applicable~~
- [ ] ~~Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes~~
- [ ] ~~Needs datastore update after deployment~~
- [x] Ready for review

### CODE REVIEW

File: frontend/javascripts/admin/dataset/dataset_upload_view.js [hunk 1]
```diff
@@ -89,6 +91,27 @@ class DatasetUploadView extends React.PureComponent<PropsWithForm, State> {
...
+          // This check has to be done because history.block triggers this function even if only the url hash changed
+          if (action === undefined || evt.pathname !== window.location.pathname) {
+            const { isUploading } = this.state;
+            if (isUploading) {
+              window.onbeforeunload = null; // clear the event handler otherwise it would be called twice. Once from history.block once from the beforeunload event
+              window.setTimeout(() => {
+                // restore the event handler in case a user chose to stay on the page
+                window.onbeforeunload = beforeUnload;
+              }, 500);
+              return messages["save.leave_page_unfinished"];
```
Comment (philippotto) [inline frontend/javascripts/admin/dataset/dataset_upload_view.js:106]: Could you create a new message string for this scenario? Right now, it says: `WARNING: You have unsaved progress that may be lost when hitting OK. Please click cancel, wait until the progress is saved and the save button displays a checkmark before leaving the page..`. However, the text doesn't fit the dataset upload page :)

**Review by philippotto — COMMENTED** [review]
Works great! Please only adapt the user message :)

File: frontend/javascripts/admin/dataset/dataset_upload_view.js (hunk 1 above)
Comment (grittaweisheit) [reply frontend/javascripts/admin/dataset/dataset_upload_view.js]: I totally forgot about that, thanks for pointing it out! It's adjusted now.

**Review by philippotto — APPROVED** [review]
:+1:

**Pull request merged**